lib/client-assets.php

```diff
@@ -591,13 +591,30 @@ function gutenberg_extend_block_editor_styles( $settings ) {
  * additions here should be complemented with a corresponding core ticket to
  * reconcile the change upstream for future removal from Gutenberg.
  *
- * @since 5.6.0
- *
- * @param array $preload_paths $preload_paths Array of paths to preload.
+ * @param array   $preload_paths Array of paths to preload.
+ * @param WP_Post $post          Post being edited.
  *
  * @return array Filtered array of paths to preload.
  */
-function gutenberg_extend_block_editor_preload_paths( $preload_paths ) {
+function gutenberg_extend_block_editor_preload_paths( $preload_paths, $post ) {
+	/*
+	 * Preload any autosaves for the post. (see https://github.com/WordPress/gutenberg/pull/7945)
+	 *
+	 * Trac ticket: https://core.trac.wordpress.org/ticket/46974
+	 *
+	 * At the time of writing, the change is not committed or released
+	 * in core. This path should be removed from Gutenberg when the code is
+	 * released in core, and the corresponding release version becomes
+	 * the minimum supported version.
+	 */
+	$post_type_object = get_post_type_object( $post->post_type );
+	$rest_base        = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
```

Comment:
While it seems unlikely given the filter flow, it makes me nervous to see `null` as a possible return type of `get_post_type_object` and then to have unguarded object property references. Should we be more guarded (wrap this whole bit in an `if ( ! is_null( $post_type_object ) ) {`)?

Reply:
👍 good idea, I've addressed that.